src/lib/navigation/base-navigation-mode-factory.provider.ts

```diff
@@ -0,0 +1,72 @@
+import {FreeNavigationMode} from './free-navigation-mode';
+import {NavigationMode} from './navigation-mode.interface';
+import {SemiStrictNavigationMode} from './semi-strict-navigation-mode';
+import {StrictNavigationMode} from './strict-navigation-mode';
+import {WizardComponent} from '../components/wizard.component';
+import {NavigationModeInput} from './navigation-mode-input.interface';
+import {NavigationModeFactory} from './navigation-mode-factory.interface';
+
+/**
+ * A factory used to create [[NavigationMode]] instances
```

Comment:
`NavigationMode` or `BaseNavigationMode` instances?

Reply:
Either :) On the one hand, generally navigation mode factories return `NavigationMode`, and all method signatures say `NavigationMode`. On the other hand, it so happens that `BaseNavigationModeFactory` returns only subclasses of `BaseNavigationMode`.
On balance, I think that `NavigationMode` is better here.